### PR TITLE
[Auto-Parallel] adapt flex_checkpoint save/load

### PR DESCRIPTION
**Pull request by Xing-lil**
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### PR types

New features
### PR changes

Others
### Description

- 适配自动并行下的flex_checkpoint，自动并行下默认使用flex_checkpoint
- 脚本中需要增加配置，判断使用hf格式或paddle格式的ckpt

```
--convert_from_hf true \
--save_to_hf true \
```

- 模型中需要添加aoa配置，具体如`modeling_network.py`中的修改
- 在 Llama-3.1-8B-Instruct 上分别验证对于 hf格式/paddle格式 的 ckpt 进行 save/load，参数/优化器状态/loss md5均能对齐

### CODE REVIEW

**Review by liym27 — APPROVED** [review]
LGTM

**Review by lugimzzz — DISMISSED** [review]
LGTM

File: paddleformers/trainer/trainer.py [hunk 1]
```diff
@@ -3436,7 +3445,7 @@ def _save_checkpoint(self, model, metrics=None):
                         OPTIMIZER_NAME: optim_state_dict,
                     }
 
-                    self._save(output_dir=os.path.join(output_dir, DIST_CKPT_PATH), state_dict=state_dict)
+                    self._save(output_dir=output_dir, state_dict=state_dict)
```
Comment (nepeplwu) [inline paddleformers/trainer/trainer.py:3448]: 这里去掉的原因是？

Comment (Xing-lil) [reply paddleformers/trainer/trainer.py:3448]: 之前自动并行会把ckpt save在output_dir下的dist_ckpt文件夹，新的flex_ckpt并不需要专门区分是否在dist_ckpt文件夹，统一保存在output_dir下即可，故删除这块旧的冗余逻辑。自动并行之后都只使用新的flex_ckpt。

**Review by nepeplwu — APPROVED** [review]
LGTM
